File: cpp/2032.two-out-of-three.cpp
#include <vector>

using std::vector;

class Solution {
public:
  vector<int> twoOutOfThree(vector<int> &nums1, vector<int> &nums2,
                            vector<int> &nums3) {

    const int n = 101;

    vector<int> set1(n, 0), set2(n, 0), set3(n, 0);

    for (int num : nums1)
      set1[num] = 1;
    for (int num : nums2)
      set2[num] = 1;
    for (int num : nums3)
      set3[num] = 1;

    vector<int> res;
    for (int i = 0; i < n; i++) {
      if (set1[i] + set2[i] + set3[i] > 1)
        res.push_back(i);
    }

    return res;
  }
};

File: python/2032.two-out-of-three.py
class Solution:
    def twoOutOfThree(
        self, nums1: list[int], nums2: list[int], nums3: list[int]
    ) -> list[int]:

        N = 101
        set1, set2, set3 = [0] * N, [0] * N, [0] * N

        for num in nums1:
            set1[num] = 1

        for num in nums2:
            set2[num] = 1

        for num in nums3:
            set3[num] = 1

        res: list[int] = []
        for i in range(1, 101):
            if set1[i] + set2[i] + set3[i] > 1:
                res.append(i)

        return res

